dbms/src/AggregateFunctions/AggregateFunctionSumMap.cpp

```diff
@@ -46,11 +56,30 @@ AggregateFunctionPtr createAggregateFunctionSumMap(const std::string & name, con
     return res;
 }
 
+AggregateFunctionPtr createAggregateFunctionSumMapFiltered(const std::string & name, const DataTypes & arguments, const Array & params)
+{
+    if (params.size() != 1)
+        throw Exception("Aggregate function " + name + "requires exactly one parameter of Array type.",
+            ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    Array keys_to_keep = params.front().safeGet<Array>();
```

Comment:
A type check for Array for better exception message?

Reply:
Done!